### PR TITLE
Simplify context in user space (part 3) 

### PR DESCRIPTION
**Pull request by pj1031999**
Use `ucontext_t` as `jmp_buf` representation.

### CODE REVIEW

File: lib/libc/gen/mips/sigsetjmp.S [hunk 1]
```diff
@@ -53,7 +52,7 @@
 
 LEAF(sigsetjmp)
 	PIC_PROLOGUE(sigsetjmp)
-	REG_S	a1, _JBLEN*SZREG(a0)		# save "savemask"
+	REG_S	a1, (a0)		        # save "savemask"
```
Comment (cahirwpz) [inline lib/libc/gen/mips/sigsetjmp.S:55]: Should not you use `uc_flags` here?

File: lib/libc/gen/mips/sigsetjmp.S [hunk 2]
```diff
@@ -62,7 +61,7 @@ END(sigsetjmp)
 
 LEAF(siglongjmp)
 	PIC_PROLOGUE(siglongjmp)
-	REG_L	t0, _JBLEN*SZREG(a0)		# get "savemask"
+	REG_L	t0, (a0)		        # get "savemask"
```
Comment (cahirwpz) [inline lib/libc/gen/mips/sigsetjmp.S:64]: Same as above.

File: lib/libc/gen/mips/sigsetjmp.S [hunk 3]
```diff
@@ -53,7 +52,7 @@
 
 LEAF(sigsetjmp)
 	PIC_PROLOGUE(sigsetjmp)
-	REG_S	a1, _JBLEN*SZREG(a0)		# save "savemask"
+	REG_S	a1, (a0)		        # save "savemask"
 	bne	a1, 0x0, 1f			# do saving of signal mask?
```
Comment (cahirwpz) [inline lib/libc/gen/mips/sigsetjmp.S:56]: Umm... did you mean `_UC_SIGMASK` ?

**Review by cahirwpz — COMMENTED** [review]
Please make sure you use `_UC_*` flags correctly. You don't have to include C header files from assembly – just put required symbol into `genassym.cf` as I did with `_UC_FPU`.

File: lib/libc/gen/mips/sigsetjmp.S (hunk 1 above)
Comment (pj1031999) [reply lib/libc/gen/mips/sigsetjmp.S]: done

File: lib/libc/gen/mips/sigsetjmp.S (hunk 2 above)
Comment (pj1031999) [reply lib/libc/gen/mips/sigsetjmp.S]: done

File: lib/libc/gen/mips/sigsetjmp.S (hunk 3 above)
Comment (pj1031999) [reply lib/libc/gen/mips/sigsetjmp.S:55]: done

**Pull request merged**